### PR TITLE
cffi: Implement wrapping of functions with slices/arrays of builtin arguments

### PR DESCRIPTION
**Pull request by corona10**
* Support Python array arguments and Go array/slice argument to pass into the function.
* Add builtin_array test. (Py2 engine does not yet pass this test.)

Updates: #128

### CODE REVIEW

**Comment by corona10** [comment]
@sbinet Please take a look when you have a time. Enjoy your weekend 👍

File: _examples/builtin_array/builtin_array.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package builtin_array
```
Comment (sbinet) [inline _examples/builtin_array/builtin_array.go]: `s/builtin_array/slices/`

we are testing slices, not arrays. (in the Go sense).
and rename the parent directory too :)

File: _examples/builtin_array/builtin_array.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+func IntSum(array []int) int {
```
Comment (sbinet) [inline _examples/builtin_array/builtin_array.go]: `s/array/slice/` or `ints`

File: _examples/builtin_array/builtin_array.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	sum := 0
+	for _, value := range array {
+		sum += value
+	}
+	return sum
+}
+
+func CreateSlice() []int {
+	var s []int
```
Comment (sbinet) [inline _examples/builtin_array/builtin_array.go]: I would just:
```go
return []int{3, 5, 6, 1}
```

File: _examples/builtin_array/builtin_array.go (hunk 1 above)
Comment (sbinet) [inline _examples/builtin_array/builtin_array.go]: also: please add a license header.

File: _examples/builtin_array/test.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from __future__ import print_function
```
Comment (sbinet) [inline _examples/builtin_array/test.py]: please add a license header.

File: main_test.go [hunk 5]
```diff
@@ -930,3 +930,15 @@ pyerrors.Div(5, 2) = 2
 `),
 	})
 }
+
+func TestBuiltinArray(t *testing.T) {
```
Comment (sbinet) [inline main_test.go]: `s/TestBuiltinArray/TestBuiltinSlice/`

**Comment by corona10** [comment]
@sbinet Updated!

File: _examples/slices/slices.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package slices
+
+func IntSum(array []int) int {
```
Comment (sbinet) [inline _examples/slices/slices.go]: `s/array/slice/` or `vals`

**Comment by corona10** [comment]
@sbinet I added it:)

**Comment by corona10** [comment]
@sbinet Then I will update it into 2 packages. :)

**Comment by corona10** [comment]
@sbinet Updated!

File: _examples/arrays/arrays.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+func IntSum(a [4]int) int {
+	sum := 0
+	for i := 0; i < len(a); i++ {
+		sum += a[i]
+	}
+	return sum
+}
+
+func CreateArrays() [4]int {
```
Comment (sbinet) [inline _examples/arrays/arrays.go]: Replace 'CreateArrays' with 'CreateArray'

File: _examples/arrays/test.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017 The go-python Authors.  All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+from __future__ import print_function
+import arrays
+
+a = [1,2,3,4]
```
Comment (sbinet) [inline _examples/arrays/test.py]: I would put the same content in the python list and in the Go array (will be less confusing if/when the test fails later on)

File: _examples/slices/test.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017 The go-python Authors.  All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+from __future__ import print_function
+import slices
+
+a = [1,2,3,4,5,6]
```
Comment (sbinet) [inline _examples/slices/test.py]: Ditto: please put the same content in the list than in the Go slice. Less confusing to debug.

File: _examples/arrays/test.py (hunk 8 above)
Comment (corona10) [inline _examples/arrays/test.py]: I unify it into [1,2,3,4]

Comment (corona10) [inline _examples/arrays/test.py]: Oh give me a second

**Comment by corona10** [comment]
@sbinet Updated

**Comment by sbinet** [comment]
thanks!

**Pull request merged**